zerver/tests/tests.py

```diff
@@ -1715,6 +1716,130 @@ def test_alert_flags(self):
+            'Welcome to Zulip',
+            'pygments.css',
+            'var page_params',
+        ]
+
+        # Keep this list sorted!!!
+        expected_keys = [
+            "alert_words",
+            "autoscroll_forever",
+            "avatar_url",
+            "bot_list",
+            "can_create_streams",
+            "cross_realm_user_emails",
+            "debug_mode",
+            "default_desktop_notifications",
+            "default_language",
+            "desktop_notifications_enabled",
+            "development_environment",
+            "domain",
+            "email",
+            "email_dict",
+            "enable_digest_emails",
+            "enable_offline_email_notifications",
+            "enable_offline_push_notifications",
+            "enter_sends",
+            "event_queue_id",
+            "first_in_realm",
+            "fullname",
+            "furthest_read_time",
+            "has_mobile_devices",
+            "have_initial_messages",
+            "initial_pointer",
+            "initial_presences",
+            "initial_servertime",
+            "is_admin",
+            "language_list",
+            "last_event_id",
+            "left_side_userlist",
+            "login_page",
+            "mandatory_topics",
+            "max_message_id",
+            "maxfilesize",
+            "muted_topics",
+            "name_changes_disabled",
+            "narrow",
+            "narrow_stream",
+            "needs_tutorial",
+            "notifications_stream",
+            "password_auth_enabled",
+            "people_list",
+            "poll_timeout",
+            "product_name",
+            "prompt_for_invites",
+            "realm_allow_message_editing",
+            "realm_create_stream_by_admins_only",
+            "realm_default_streams",
+            "realm_emoji",
+            "realm_filters",
+            "realm_invite_by_admins_only",
+            "realm_invite_required",
+            "realm_message_content_edit_limit_seconds",
+            "realm_name",
+            "realm_restricted_to_domain",
+            "referrals",
+            "show_digest_email",
+            "sounds_enabled",
+            "staging",
+            "stream_desktop_notifications_enabled",
+            "stream_sounds_enabled",
+            "subbed_info",
+            "test_suite",
+            "twenty_four_hour_time",
+            "unread_count",
+            "unsubbed_info",
+            "voyager",
+        ]
+
+        email = "hamlet@zulip.com"
+
+        # Verify fails if logged-out
+        result = self.client.get('/')
+        self.assertEqual(result.status_code, 302)
+
+        # Verify succeeds once logged-in
+        self.login(email)
+        with \
+                patch('zerver.lib.actions.request_event_queue', return_value=42), \
+                patch('zerver.lib.actions.get_user_events', return_value=[]):
+            result = self.client.get('/', dict(stream='Denmark'))
+        html = str(result)
```

Comment:
Use `html = result.content.decode('utf-8')`.

`str(result)` gives `"<django.http.response.HttpResponse object at 0x7f507b39eba8>"` in python 3 instead of the response's body.